java-frontend/src/main/java/org/sonar/java/filters/EclipseI18NFilter.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.filters;
+
+import com.google.common.collect.ImmutableSet;
+
+import org.sonar.plugins.java.api.tree.ClassTree;
+
+import java.util.Set;
+
+public class EclipseI18NFilter extends BaseTreeVisitorIssueFilter {
+
+  @Override
+  public Set<String> targetedRules() {
```

Comment:
I would really rely on class name and use the AnnotationUtils class : 
```
org.sonar.check.Rule ruleAnnotation = AnnotationUtils.getAnnotation(clazz, org.sonar.check.Rule.class);
``` 
to retrieve the key and contruct this set of string. 

This way, we prevent ourselves of key change that are bound to happen and there is only one place where the key is defined.